drivers/display/display_st7735r.c

```diff
@@ -268,7 +269,16 @@ static void st7735r_get_capabilities(const struct device *dev,
 	memset(capabilities, 0, sizeof(struct display_capabilities));
 	capabilities->x_resolution = config->width;
 	capabilities->y_resolution = config->height;
-	if (config->madctl & ST7735R_MADCTL_BGR) {
+
+	/*
+	 * Invert the pixel format if rgb_is_inverted is enabled.
+	 * Report pixel format as the same format set in the MADCTL
+	 * if disabling the rgb_is_inverted option.
+	 * Or not so, reporting pixel format as RGB if MADCTL setting
+	 * is BGR. And also vice versa.
+	 * It is a workaround for supporting buggy modules that display RGB as BGR.
```

Comment:
I think it should be described differently. Do you have a link to the display's errata?

Reply:
@jfischer-no 
Sorry for my late response. 

I looked it up, but there was no mention of the errata.

#44796 supports the LCD module that built-in Longan Nano Dev-Board.

Dev-Board Manufacturer(sipeed.com) provides these documents, 

Dev-Board datasheet
https://dl.sipeed.com/fileList/LONGAN/Nano/Spec/Sipeed%20longan%20nano%20Datasheet%20V1.0.pdf

LCD module datasheet
https://dl.sipeed.com/fileList/LONGAN/Nano/HDK/0.96%20inch%2080x160%20IPS%20LCD.pdf

But, no mention is in these documents about errata.

LCD module manufacturer's page also has no good information about the module.

https://www.swicn.com/product-by-category/6

(I feel the manufacturer considers it is internal specs, not a bug.
They did not mention about ST7735 settings.)

Reply:
@mbolivar-nordic @soburi Would it not be better to handle it using specific compatible? Something like
`
compatible: "longan-nano,st7735r"`

```
#if DT_HAS_COMPAT_STATUS_OKAY(longan_nano_st7735r)
     apply_fix_for_longan_nano();
#endif
```

Reply:
@jfischer-no 
I agree this workaround enable only in the case the by by specific compatible.
I'll try rework it.

Reply:
Please wait for @mbolivar-nordic feedback :-)

Reply:
I noticed a problem.
before comma part of compatible name must be a vender prefix... :-(

Reply:
There is dts/bindings/vendor-prefixes.txt

Reply:
I think it is not good idea. longan_nano is board name. I think should not be in vender-prefixs.txt
If replace longan_nano by that manufacturer name(sipeed,st7735r.yaml), 
this look like all of their product that use st7735r may affect this workaround... :-(

Reply:
longan-nano was just an example

Reply:
I can't think good alternative yet.
I also wait @mbolivar-nordic feedback.

Reply:
Thank you for your detailed description of the problem @soburi. It seems like we are dealing with a device "quirk". I think a simple property is probably OK if a simple "inverted" is the only quirk you ever expect to see for a display handled by this controller.

However, I wonder if you might have other problems with other vendors. For example, if MADCTL says the color order is RGB, but the actual display order from some vendor is GRB. If that happens, `rgb-is-inverted` will not be enough. Do you think this is possible?

If so, you might want to define a `quirky-display-color-order` property that lets you define the actual display color order instead. For my example, you could write something like `quirky-display-color-order = <GREEN>, <RED>, <BLUE>;`. 

This is basically what we have to do in LED and LED strip bindings like this one:

https://docs.zephyrproject.org/latest/build/dts/api/bindings/led_strip/worldsemi%2Cws2812-spi.html#dtbinding-worldsemi-ws2812-spi

See the `color-mapping` property for details and the `drivers/led_strip/ws2812_spi.c` driver for how it's handled for that device.

In summary, if you think the current `rgb-is-inverted` property will be general enough to handle all vendor quirks, then it is OK for me. If you think vendors will be "creative" and pick any color order sometimes, you may want to consider a more general property.

I leave it to you to decide; either approach seems fine to me from the DTS side.

Reply:
Hi, @mbolivar-nordic,

> For example, if MADCTL says the color order is RGB, but the actual display order from some vendor is GRB. If that happens, rgb-is-inverted will not be enough. Do you think this is possible?

I think it cannot. MADCTL is binary property. It can select only from RGB or BGR format.

By the way, ST7735 also has an 'RGB' inverting pin' that control by wire.
From it, We can say that ST7735 considers RGB/BGR inverting but not considering other pixel formats. (Of course, It does not mention other pixel formats in datasheets.)

So I think 'rgb-is-inverted' property is enough to support functions of ST7735.

Reply:
> So I think 'rgb-is-inverted' property is enough to support functions of ST7735.

Very well, that sounds good to me as well. Thank you for the explanation.